net: lib: nrf_cloud: UUID as the default device ID (#28845)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/migration/migration_guide_3.4.rst

This section describes the changes related to libraries.

nRF Cloud library
-----------------

.. toggle::

   * The default value of the :kconfig:option:`CONFIG_NRF_CLOUD_CLIENT_ID_SRC` choice has changed from :kconfig:option:`CONFIG_NRF_CLOUD_CLIENT_ID_SRC_IMEI` to the device UUID.

>      On nRF9160 the new default is :kconfig:option:`CONFIG_NRF_CLOUD_CLIENT_ID_SRC_INTERNAL_UUID`; on nRF91x1 and future SoCs the new default is :kconfig:option:`CONFIG_NRF_CLOUD_CLIENT_ID_SRC_MDM_DEVICE_UUID`.

Reviewer: ```suggestion
     On an nRF9160 device, the new default is :kconfig:option:`CONFIG_NRF_CLOUD_CLIENT_ID_SRC_INTERNAL_UUID`
     On the nRF91x1 and future SoCs, the new default is :kconfig:option:`CONFIG_NRF_CLOUD_CLIENT_ID_SRC_MDM_DEVICE_UUID`.
```

>      If your application relied on the previous IMEI-based default without setting it explicitly in :file:`prj.conf`, the device will connect to nRF Cloud with a different client ID after the upgrade and appear as a new, unprovisioned device.

Reviewer: ```suggestion
     If your application relied on the previous IMEI-based default without setting it explicitly in the :file:`prj.conf` file, the device will connect to nRF Cloud with a different client ID after the upgrade and appear as a new, unprovisioned device.
```

     To preserve the previous behavior, add the following line to your :file:`prj.conf` file:

     .. code-block:: cfg

        CONFIG_NRF_CLOUD_CLIENT_ID_SRC_IMEI=y

     :kconfig:option:`CONFIG_NRF_CLOUD_CLIENT_ID_SRC_IMEI` is deprecated but kept functional for fleet operators with IMEI-provisioned devices already in the field.

>      :kconfig:option:`CONFIG_NRF_CLOUD_CLIENT_ID_SRC_HW_ID` is also deprecated and kept functional for the same reason.

Reviewer: ```suggestion
     The :kconfig:option:`CONFIG_NRF_CLOUD_CLIENT_ID_SRC_IMEI` and :kconfig:option:`CONFIG_NRF_CLOUD_CLIENT_ID_SRC_HW_ID` Kconfig options are deprecated but kept functional for fleet operators with IMEI-provisioned devices already in the field.
```
